From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tamer Ahmed <tamerahmed@microsoft.com>
Date: Fri, 26 Jun 2026 17:38:00 +0000
Subject: [PATCH] SONiC-ONLY: build SONiC FRR Redis interface

Build the SONiC FRR Redis interface as a standalone zebra dataplane
module. The module publishes EVPN-MH dataplane state directly to APP_DB
and resolves Redis DB configuration through swss-common.

Signed-off-by: Tamer Ahmed <tamerahmed@microsoft.com>
---
 debian/control | 4 +++-
 debian/frr.install | 1 +
 zebra/subdir.am | 6 ++++++
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/debian/control b/debian/control
index 40f820ae44..b5012a7fa8 100644
--- a/debian/control
+++ b/debian/control
@@ -41,3 +41,5 @@ Build-Depends: bison,
                libprotobuf-dev (>=3.6.1) <pkg.frr.grpc>,
-               libgoogle-perftools-dev <pkg.frr.tcmalloc>
+               libgoogle-perftools-dev <pkg.frr.tcmalloc>,
+               libhiredis-dev,
+               libswsscommon-dev
 Standards-Version: 4.5.0.3
diff --git a/debian/frr.install b/debian/frr.install
index 375f0dadd7..97bd5723d2 100644
--- a/debian/frr.install
+++ b/debian/frr.install
@@ -14,2 +14,3 @@ usr/lib/*/frr/modules/dplane_fpm_nl.so
 usr/lib/*/frr/modules/dplane_fpm_sonic.so
+usr/lib/*/frr/modules/sonic_frr_redis_interface.so
 usr/lib/*/frr/modules/zebra_cumulus_mlag.so
diff --git a/zebra/subdir.am b/zebra/subdir.am
index 9b86d2c5b7..4734092bb7 100644
--- a/zebra/subdir.am
+++ b/zebra/subdir.am
@@ -253,2 +253,8 @@ zebra_dplane_fpm_sonic_la_LIBADD  = -lnexthopgroup
-
+
+module_LTLIBRARIES += zebra/sonic_frr_redis_interface.la
+
+zebra_sonic_frr_redis_interface_la_SOURCES = zebra/sonic_frr_redis_interface.c
+zebra_sonic_frr_redis_interface_la_LDFLAGS = $(MODULE_LDFLAGS)
+zebra_sonic_frr_redis_interface_la_LIBADD  = -lhiredis -lswsscommon
+
 if NETLINK_DEBUG
